endpoints/api/namespacequota.py

```diff
@@ -75,252 +74,322 @@ class OrganizationQuota(ApiResource):
         try:
-            model.namespacequota.create_namespace_limit(
-                orgname=namespace,
-                percent_of_limit=quota_limit_data["percent_of_limit"],
-                quota_type_id=quota_limit_data["quota_type_id"],
+            model.namespacequota.create_namespace_quota_limit(
+                quota,
+                quota_type,
+                quota_limit_threshold,
             )
             return "Created", 201
         except model.DataModelException as ex:
             raise request_error(exception=ex)
 
-    @nickname("changeOrganizationQuotaLimit")
-    @validate_json_request("NewOrgQuotaLimit")
-    def put(self, namespace):
 
-        superperm = SuperUserPermission()
+@resource("/v1/organization/<orgname>/quota/<quota_id>/limit/<limit_id>")
+@show_if(features.QUOTA_MANAGEMENT)
+class OrganizationQuotaLimit(ApiResource):
+    schemas = {
+        "UpdateOrgQuotaLimit": {
+            "type": "object",
+            "description": "Description of changing organization quota limit",
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "description": 'Type of quota limit: "Warning" or "Reject"',
+                },
+                "threshold_percent": {
+                    "type": "integer",
+                    "description": "Quota threshold, in percent of quota",
+                },
+            },
+        },
+    }
 
-        if not superperm.can():
+    @nickname("getOrganizationQuotaLimit")
+    def get(self, orgname, quota_id, limit_id):
+        orgperm = OrganizationMemberPermission(orgname)
+        if not orgperm.can():
             raise Unauthorized()
 
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+        quota_limit = model.namespacequota.get_namespace_quota_limit(quota, limit_id)
+        if quota_limit is None:
+            raise NotFound()
+
+        return limit_view(quota_limit)
+
+    @nickname("changeOrganizationQuotaLimit")
+    @validate_json_request("UpdateOrgQuotaLimit")
+    def put(self, orgname, quota_id, limit_id):
+        orgperm = AdministerOrganizationPermission(orgname)
+
+        if not features.SUPER_USERS or not SuperUserPermission().can():
+            if not orgperm.can():
+                raise Unauthorized()
+
         quota_limit_data = request.get_json()
 
-        try:
-            quota_limit_id = quota_limit_data["quota_limit_id"]
-        except KeyError:
-            msg = "Must supply quota_limit_id for updates"
-            raise request_error(message=msg)
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+        quota_limit = model.namespacequota.get_namespace_quota_limit(quota, limit_id)
+        if quota_limit is None:
+            raise NotFound()
 
-        quota = model.namespacequota.get_namespace_limit_from_id(namespace, quota_limit_id)
+        if "type" in quota_limit_data:
```

Comment:
validation at start to ensure we got at least 1 parameter to avoid running through this. Potentially return an error with invalid request configuration.